src/lazy/binary/raw/v1_1/struct.rs

```diff
@@ -215,13 +241,23 @@ impl<'top> RawBinaryStructIterator_1_1<'top> {
     /// Helper function called from [`Self::next`] to parse the current field and value from the
     /// struct. On success, returns both the field pair via [`LazyRawFieldExpr`] as well as the
     /// total bytes needed to skip the field.
-    fn peek_field(&self) -> IonResult<Option<(LazyRawFieldExpr<'top, BinaryEncoding_1_1>, usize)>> {
-        let mut buffer = self.source;
+    fn peek_field(
+        &mut self,
+        input: ImmutableBuffer<'top>,
+    ) -> IonResult<Option<(LazyRawFieldExpr<'top, BinaryEncoding_1_1>, usize)>> {
+        let mut buffer = input;
```

Comment:
Could/should this method return a `ParseResult<LazyRawFieldExpr<_>>` instead so the caller gets the remaining input? (And can deduce the length as desired.)

EDIT: I see now that this is being used to populate the offsets cache. I think this is another place where it would make sense to store the `LazyRawFieldExpr` in the cache instead of just the offset to avoid later rework.